core/controllers/reader_test.py

```diff
@@ -188,6 +193,51 @@ def test_give_feedback_handler(self):
         self.logout()
 
 
+class ExplorationHandlerTest(test_utils.GenericTestBase):
+    """Test the handler for initialising exploration with
+    state_classifier_mapping.
+    """
+
+    def test_creation_of_state_classifier_mapping(self):
+        super(ExplorationHandlerTest, self).setUp()
+        exploration_id = '15'
+
+        self.login(self.VIEWER_EMAIL)
+        self.signup(self.VIEWER_EMAIL, self.VIEWER_USERNAME)
+
+        # Load demo exploration.
+        exp_services.delete_demo(exploration_id)
+        with self.swap(feconf, 'ENABLE_ML_CLASSIFIERS', True):
```

Comment:
@seanlip load_demo() calls save_exploration_from_yaml() method. Only if ML classifiers are enabled, that function will lead to training job creation for trainable states.